cl_manycore/libraries/bsg_manycore_mem.cpp

```diff
@@ -121,8 +121,10 @@ static int hb_mc_npa_is_dram (npa_t *npa) {
  * checks if NPA is in host endpoint.
  */
 static int hb_mc_npa_is_host (npa_t *npa) {
-	uint8_t dim_x = hb_mc_get_manycore_dimension_x();
-	if ((npa->y == 0) && (npa->x == dim_x))
+	uint8_t coord_x = hb_mc_get_host_intf_coord_x();
```

Comment:
Can you use the hb_mc_idx_t type instead of uint8_t?

Reply:
These functions are deprecated, which is why I've left them with the deprecated get_host_intf_coord() function (it returns a uint8_t so I didn't change the type). I'll add deprecated tags to all of them.

Reply:
Right on